rust/README.md

```diff
@@ -0,0 +1,10 @@
+# Rust development guide
+
+Firezone uses Rust for all data plane components. This directory contains the
+Linux and Windows clients, and low-level networking implementations related to
+STUN/TURN.
+
+## Developer Setup
+
+- Install rustup from https://rustup.rs/
+- Run `cargo --version` to make sure you have Rust installed.
```

Comment:
YMMV but not everybody uses rustup (I don't for example) so it would be more correct / helpful to say:

```suggestion
We target the last stable release of Rust using [`rust-toolchain.toml`](./rust-toolchain.toml).
If you are using `rustup`, that is automatically handled for you.
Otherwise, ensure you have the latest stable version of Rust installed.
```

Reply:
Huh. Do you install Cargo with Nix or something?

Reply:
I mentioned rustup caused I figured it's what novices use, and anyone who doesn't know rustup knows more than the README file anyway

Reply:
> Huh. Do you install Cargo with Nix or something?

Yeah I have an overlay where I can specify, which Rust version I want and it is set to latest stable. I haven't yet gotten around to setting up dev-shells properly but the idea would be that I can control per dev-shell which version I want.

Reply:
> I mentioned rustup caused I figured it's what novices use, and anyone who doesn't use rustup knows more than the README file anyway

Yeah that makes sense. I guess an expert can also find the rust-toolchain file by themselves :)

Reply:
Looks like asdf can handle it too. I just started trying it a day or two ago after noticing it in the CONTRIBUTING.md doc